src/borg/remote.py

```diff
@@ -646,8 +645,8 @@ def borg_cmd(self, args, testing):
             except AttributeError:
                 pass
         env_vars = []
-        if yes(env_var_override='BORG_HOSTNAME_IS_UNIQUE', env_msg=None, prompt=False):
-            env_vars.append('BORG_HOSTNAME_IS_UNIQUE=yes')
+        if not hostname_is_unique():
+            env_vars.append('BORG_HOSTNAME_IS_UNIQUE=no')
```

Comment:
Just to clarify: The whole SSH_ORIGINAL_COMMAND enchilada only matters iff an SSH forced command is used. Otherwise this directly works, since the client has direct shell access.